src/plots/plot.ts

```diff
@@ -374,16 +353,50 @@ export class Plot extends Component {
   }
 
   /**
+   * Adds a Dataset to the Plot.
+   *
+   * @param {Dataset} dataset
+   * @returns {Plot} The calling Plot.
+   */
+  public addDataset(dataset: Dataset) {
+    return this._addDataset(dataset, true);
+  }
+
+  private _addDataset(dataset: Dataset, dispatchUpdate: boolean) {
+    this._removeDataset(dataset, false);
+    let drawer = this._createDrawer(dataset);
+    this._datasetToDrawer.set(dataset, drawer);
+
+    if (this._isSetup) {
+      this._createNodesForDataset(dataset);
+    }
+
+    dataset.onUpdate(this._onDatasetUpdateCallback);
+    if (dispatchUpdate) {
+      this._onDatasetUpdate();
+    }
+    return this;
+  }
+
+  /**
    * Removes a Dataset from the Plot.
    *
    * @param {Dataset} dataset
    * @returns {Plot} The calling Plot.
    */
   public removeDataset(dataset: Dataset): Plot {
-    if (this.datasets().indexOf(dataset) > -1) {
-      this._removeDatasetNodes(dataset);
-      dataset.offUpdate(this._onDatasetUpdateCallback);
-      this._datasetToDrawer.delete(dataset);
+    return this._removeDataset(dataset, true);
+  }
+
+  private _removeDataset(dataset: Dataset, dispatchUpdate: boolean) {
```

Comment:
Instead of having a boolean parameter here, can we have this function not do any updating and then have `removeDataset()` do the updating manually?